avocado/core/job.py

```diff
@@ -556,6 +563,8 @@ def run(self):
             self.log.error('Report bugs visiting %s', _NEW_ISSUE_LINK)
             return exit_codes.AVOCADO_FAIL
         finally:
+            if self.job_post_dispatcher.extensions:
+                self.job_post_dispatcher.map_method('run', self)
```

Comment:
This looks a bit too granular to me. How about implementing only a generic `Job` plugin, which would let you use one or more hooks. So instead of reserving:

`avocado.plugins.job.*`

you'd reserve only `avocado.plugins.job` and instead of `run` you'd implement one or more hooks:

```python
class JobPrePost(plugin):
    def __init__(self, job):    # I'm not sure this is possible
        # if not we can pass the job reference each time instead
        super(JobPrePost, self).__init__()
        self.job = job

    def pre(self):   # (self, job) if it's not possible to get it on __init__ time
        # if defined runs before the `job.run`

    def post(self):
        # if defined runs after the `job.run`
```

Eventually we'd add more hooks for finer control of the job flow. It'd be easier to program this to me as for example extended sleep example could look like this:

```python
import time
import logging

from avocado.core.settings import settings
from avocado.plugins.base import Job


class Sleep(Job):

    name = 'sleep'
    description = 'Sleeps for a number of seconds'

    def _run(self, job, event):
        log = logging.getLogger("avocado.app")
        seconds = settings.get_value(section="plugins.job.sleep",
                                     key=event,
                                     key_type=int,
                                     default=3)
        for i in xrange(1, seconds + 1):
            log.info("Sleeping %s job %2i/%s", event, i, seconds)
            time.sleep(1)

    def pre(self, job):
        self._run(job, "pre")

    def post(self, job):
        self._run(job, "post")
```